packages/tests/test/reya_common/trade/PassivePool.fork.c.sol

```diff
@@ -255,4 +266,288 @@ contract PassivePoolForkCheck is BaseReyaForkTest {
+
+        // user executes short trade on ETH
+        executeCoreMatchOrder({ marketId: 1, sender: user, base: sd(-10e18), priceLimit: ud(0), accountId: accountId });
+
+        // user closes the short trade on ETH and goes same long
+        executeCoreMatchOrder({
+            marketId: 1,
+            sender: user,
+            base: sd(20e18),
+            priceLimit: ud(type(uint256).max),
+            accountId: accountId
+        });
+
+        // withdraw 100 deUSD from account
+        executePeripheryWithdrawMA(user, userPk, 1, accountId, sec.sdeusd, 100e18, sec.mainChainId);
+    }
+
+    function autoRebalancePool() internal {
+        removeCollateralWithdrawalLimit(sec.rusd);
+        removeCollateralWithdrawalLimit(sec.deusd);
+        removeCollateralWithdrawalLimit(sec.sdeusd);
+
+        address quoteToken = sec.rusd;
+        address[] memory supportingTokens = IPassivePoolProxy(sec.pool).getQuoteSupportingCollaterals(sec.passivePoolId);
+
+        address[] memory allTokens = new address[](supportingTokens.length + 1);
+        allTokens[0] = quoteToken;
+        for (uint256 i = 0; i < supportingTokens.length; i++) {
+            allTokens[i + 1] = supportingTokens[i];
+        }
+
+        for (uint256 i = 0; i < allTokens.length; i++) {
+            for (uint256 j = 0; j < allTokens.length; j++) {
+                if (i == j) {
+                    continue;
+                }
+
+                address tokenIn = allTokens[i];
+                address tokenOut = allTokens[j];
+
+                uint256 amountIn = 100_000_000 * (10 ** IERC20TokenModule(tokenIn).decimals());
+
+                RebalanceAmounts memory rebalanceAmounts =
+                    IPassivePoolProxy(sec.pool).getRebalanceAmounts(sec.passivePoolId, tokenIn, tokenOut, amountIn);
+
+                if (rebalanceAmounts.amountIn != 0) {
+                    vm.prank(dec.socketController[tokenIn]);
+                    IERC20TokenModule(tokenIn).mint(sec.rebalancer1, rebalanceAmounts.amountIn);
+
+                    vm.prank(sec.rebalancer1);
+                    IERC20TokenModule(tokenIn).approve(sec.pool, rebalanceAmounts.amountIn);
+
+                    vm.prank(sec.rebalancer1);
+                    IPassivePoolProxy(sec.pool).triggerAutoRebalance(
+                        sec.passivePoolId,
+                        AutoRebalanceInput({
+                            tokenIn: tokenIn,
+                            amountIn: rebalanceAmounts.amountIn,
+                            tokenOut: tokenOut,
+                            minPrice: rebalanceAmounts.priceInToOut,
+                            receiverAddress: sec.periphery
+                        })
+                    );
+
+                    vm.warp(block.timestamp + 1);
+                }
+            }
+        }
+    }
+
+    function check_autoRebalance_currentTargets() public {
+        autoRebalancePool();
+    }
+
+    function check_autoRebalance_differentTargets() public {
```

Comment:
hmm, i think it'b be nice to have access restriction checks as well (not sure if we have them somewhere already)

- only authorized rebalancers can rebalance and nobody else
- only authorised owners can leverage v2 liquidity add and remove operations (note, owners rather than msg.senders, check out the latest change in the passive pool implementation)